BaragonData/src/main/java/com/hubspot/baragon/data/AbstractDataStore.java

```diff
@@ -152,14 +153,14 @@ public T apply(byte[] data) {
     }
   }
 
-  protected <T> T deserialize(byte[] data, Class<T> klass) {
+  protected <T> T deserialize(byte[] data, Class<T> klass, String path) {
     try {
       return objectMapper.readValue(data, klass);
     } catch (JsonParseException jpe) {
       try {
-        LOG.error(String.format("Invalid Json: %s", objectMapper.readValue(data, String.class)));
+        LOG.error("Invalid Json at path {}: {}", path, new String(data, StandardCharsets.UTF_8), jpe);
       } catch (Exception e) {
-        LOG.error("Could not get raw json string", e);
+        LOG.error("Could not parse valid json at path {}", path, e);
```

Comment:
If odd characters somehow got into the String and we can't convert the bytes to utf-8 it will still throw an exception here (We were actually hitting that exception block before, whatever odd stuff was in the json was something object mapper couldn't convert to a string, thus the new String()). Not sure how invalid chars would get in there. But not sure how the invalid json got in there necessitating this either.